### PR TITLE
sys-apps/diffutils-3.5: build with gcc-7.1 fix

### PR DESCRIPTION
**Pull request by lekto**
Gentoo-Bug: https://bugs.gentoo.org/617538

### CODE REVIEW

File: sys-apps/diffutils/diffutils-3.5.ebuild [hunk 1]
```diff
@@ -20,6 +20,12 @@ DEPEND="app-arch/xz-utils
 
 DOCS=( AUTHORS ChangeLog NEWS README THANKS TODO )
 
+src_unpack() {
+	unpack ${A}
+	cd "${S}"
+	epatch "${FILESDIR}/${P}-fix_macros.patch"
```
Comment (SoapGentoo) [inline sys-apps/diffutils/diffutils-3.5.ebuild]: this is the wrong fix. Please use `src_prepare` for patching.

**Comment by mgorny** [comment]
Was the patch upstreamed? Is there really a point in backporting it when we're far even from gcc6 in mainstream Gentoo? I'd say it's quite likely a new release will happen even before we get gcc7 in ::gentoo.

**Comment by lekto** [comment]
It's fixed in upstream in master branch. Gcc-6 is quite stable for me already (one computer and one laptop) and update to gcc-7 broke much less then update form gcc-5 to gcc-6. So I think gentoo go much quicker from gcc-6 to gcc-7, then gcc-5 to gcc-6.

File: sys-apps/diffutils/diffutils-3.5.ebuild (hunk 1 above)
Comment (lekto) [inline sys-apps/diffutils/diffutils-3.5.ebuild]: fixed in 33dbb39

**Pull request closed without merging**